group.go

```diff
@@ -146,9 +146,8 @@ func (g *group) Go(op func(context.Context)) {
 }
 
 func (g *group) Wait() {
-	if !g.awaited.Swap(true) {
-		g.wg.Wait()
-	}
+	g.awaited.Store(true)
```

Comment:
Makes sense always call wg.Wait here. 

And we now Store instead of Swap, because we don't actually care if we have already called Wait?

Reply:
yeah Swap does basically the same thing but it returns the old value too. if we aren't looking at it it doesn't matter, we just want the write fence.

Reply:
i think i may have either:
1. had a misconception that waitgroups block when awaited again or something, or
2. gotten confuddled between the simple group and the more complex groups which *do* need a once-only shutdown action because they have channels, workers, and finalizers that must be stopped.